typescript/lib/parsers/openai.ts

```diff
@@ -645,9 +658,16 @@ export class OpenAIChatModelParser extends ParameterizedModelParser<Chat.ChatCom
     }
 
     if (output.output_type === "execute_result") {
-      // TODO: Add in OutputDataWithValue another way to support function calls
-      if (typeof output.data === "string") {
+      if (output.data?.hasOwnProperty("value")) {
+        const outputData = output.data as OutputDataWithValue;
+        if (typeof outputData.value === "string") {
+          return outputData.value;
+        }
+        return JSON.stringify(outputData.value); // function_call
+      } else if (typeof output.data === "string") {
         return output.data;
+      } else if (output.metadata?.rawResponse?.function_call) {
+        return JSON.stringify(output.metadata?.rawResponse?.function_call);
       }
```

Comment:
What is the case where we need this? Should be safe to remove since line 661-667 should handle this, right?

Reply:
Thanks for the catch, I just missed it from all the if-else overloading my brain